build-utils/integration-docs-fetch-plugin.js

```diff
@@ -0,0 +1,76 @@
+/*eslint-env node*/
+/*eslint import/no-nodejs-modules:0 */
+const fs = require('fs');
+const https = require('https');
+const path = require('path');
+
+const PLATFORMS_URL = 'https://docs.sentry.io/_platforms/_index.json';
+const DOCS_INDEX_PATH = 'src/sentry/integration-docs/_platforms.json';
+
+const alphaSortFromKey = function(keyExtractor) {
+  return function(a, b) {
+    const nameA = keyExtractor(a);
+    const nameB = keyExtractor(b);
+    return nameA.localeCompare(nameB);
+  };
+};
+
+const transformPlatformsToList = ({platforms}) =>
+  Object.keys(platforms)
+    .map(platformId => {
+      const integrationMap = platforms[platformId];
+      const integrations = Object.keys(integrationMap)
+        .sort(alphaSortFromKey(key => integrationMap[key].name))
+        .map(integrationId => {
+          const {name, type, doc_link: link} = integrationMap[integrationId];
+          const id =
+            integrationId === '_self' ? platformId : `${platformId}-${integrationId}`;
+          return {
+            id,
+            name,
+            type,
+            link,
+          };
+        });
+      return {
+        id: platformId,
+        name: integrationMap._self.name,
+        integrations,
+      };
+    })
+    .sort(alphaSortFromKey(item => item.name));
+
+class IntegrationDocsFetchPlugin {
+  constructor({basePath}) {
+    this.modulePath = path.join(basePath, DOCS_INDEX_PATH);
+  }
+
+  apply(compiler) {
+    compiler.hooks.beforeRun.tapAsync(
+      'IntegrationDocsFetchPlugin',
+      (compilation, callback) => {
+        https
+          .get(PLATFORMS_URL, res => {
+            res.setEncoding('utf8');
+            let buffer = '';
+            res
+              .on('data', data => {
+                buffer += data;
+              })
+              .on('end', () =>
+                fs.writeFile(
```

Comment:
Can we check if the file already exists and skip if it does?

Reply:
I thought about it and decided against as the file maybe stale. We can use the `if-modified-since` header with file's timestamp but I'd defer that to a follow up diff as this is merely replicating what Python does. WDYT?

Reply:
Yeah that's fine 👍